src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +31,263 @@ public void GetSerializableMembers_Interface()
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_NetCore_TestData))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_NetCore_ThrowsNotSupportedException(Type type)
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The coreclr doesn't support GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_Netfx_ThrowsNotSupportedException(Type type)
+        {
+            Assert.NotNull(FormatterServices.GetUninitializedObject(type));
+            Assert.NotNull(FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Netcoreapp, "The coreclr doesn't support GetUninitializedObject for shared generic instances")]
+        public void GetUninitializedObject_SharedGenericInstance_NetCore_ThrowsNotSupportedException()
+        {
+            Type canonType = Type.GetType("System.__Canon");
+            Assert.NotNull(canonType);
+            Type sharedGenericInstance = typeof(GenericClass<>).MakeGenericType(canonType);
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(sharedGenericInstance));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(sharedGenericInstance));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The full .NET framework partially supports GetUninitializedObject for shared generic instances")]
+        public void GetUninitializedObject_SharedGenericInstance_Netfx_InitializesValue()
+        {
+            Type canonType = Type.GetType("System.__Canon");
```

Comment:
Right, it won't work. I had the same comment above that it is questionable for corefx tests to depend on internal CoreLib types.